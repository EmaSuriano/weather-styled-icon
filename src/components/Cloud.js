import styled from 'styled-components';
import Pan from './animations/Pan';
import { ThemePropType, defaultTheme } from '../constants';

const Cloud = styled.div`
  position: absolute;
  z-index: 1;
  top: 50%;
  left: 50%;
  width: 3.6875em;
  height: 3.6875em;
  margin: -1.84375em;
  background: ${({ theme }) => theme.backgroundColor};
  border-radius: 50%;
  box-shadow: -2.1875em 0.6875em 0 -0.6875em ${({ theme }) => theme.backgroundColor},
    2.0625em 0.9375em 0 -0.9375em ${({ theme }) => theme.backgroundColor},
    0 0 0 0.375em ${({ theme }) => theme.cloudsColor},
    -2.1875em 0.6875em 0 -0.3125em ${({ theme }) => theme.cloudsColor},
    2.0625em 0.9375em 0 -0.5625em ${({ theme }) => theme.cloudsColor};

  &:after {
    content: '';
    position: absolute;
    bottom: 0;
    left: -0.5em;
    display: block;
    width: 4.5625em;
    height: 1em;
    background: ${({ theme }) => theme.backgroundColor};
    box-shadow: 0 0.4375em 0 -0.0625em ${({ theme }) => theme.cloudsColor};
  }

  &:nth-child(2) {
    z-index: 0;
    background: ${({ theme }) => theme.cloudsColor};
    box-shadow: -2.1875em 0.6875em 0 -0.6875em ${({ theme }) => theme.cloudsColor},
      2.0625em 0.9375em 0 -0.9375em ${({ theme }) => theme.cloudsColor},
      0 0 0 0.375em ${({ theme }) => theme.cloudsColor},
      -2.1875em 0.6875em 0 -0.3125em ${({ theme }) => theme.cloudsColor},
      2.0625em 0.9375em 0 -0.5625em ${({ theme }) => theme.cloudsColor};
    opacity: 0.6;
    transform: scale(0.5) translate(6em, -3em);
    animation: ${Pan} 4s linear infinite;
  }

  &:nth-child(2):after {
    background: ${({ theme }) => theme.cloudsColor};
  }
`;

Cloud.displayName = 'Cloud';

Cloud.propTypes = {
  theme: ThemePropType,
};

Cloud.defaultProps = {
  theme: defaultTheme,
};

export default Cloud;
