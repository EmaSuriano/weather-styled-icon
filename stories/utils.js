import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { WeatherThemeProvider } from '../src';
import { ThemePropType } from '../src/constants';

export const CenterContainer = styled.div`
  ${props =>
    props.theme.backgroundColor &&
    `background-color: ${props.theme.backgroundColor}`};
  text-align: center;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const BackgroundContainer = styled(CenterContainer)`
  background-color: ${props => props.color};
`;

export const ThemeContainer = ({ theme, children }) => (
  <WeatherThemeProvider theme={theme}>
    <BackgroundContainer>{children}</BackgroundContainer>
  </WeatherThemeProvider>
);

ThemeContainer.propTypes = {
  theme: ThemePropType.isRequired,
  children: PropTypes.node,
};

ThemeContainer.defaultProps = {
  children: PropTypes.node.isRequired,
};
