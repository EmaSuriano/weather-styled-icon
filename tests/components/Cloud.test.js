import React from 'react';
import Cloud from '../../src/components/Cloud';
import { renderWithTheme } from '../utils';

describe('<Cloud />', () => {
  it('should should render div with theme', () => {
    const tree = renderWithTheme(<Cloud />);
    expect(tree).toMatchSnapshot();
  });
});
