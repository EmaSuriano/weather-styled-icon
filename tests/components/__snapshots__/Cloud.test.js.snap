// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cloud /> should should render div with theme 1`] = `
.c0 {
  position: absolute;
  z-index: 1;
  top: 50%;
  left: 50%;
  width: 3.6875em;
  height: 3.6875em;
  margin: -1.84375em;
  background: #FFFFFF;
  border-radius: 50%;
  box-shadow: -2.1875em 0.6875em 0 -0.6875em #FFFFFF, 2.0625em 0.9375em 0 -0.9375em #FFFFFF, 0 0 0 0.375em #607D8B, -2.1875em 0.6875em 0 -0.3125em #607D8B, 2.0625em 0.9375em 0 -0.5625em #607D8B;
}

.c0:after {
  content: '';
  position: absolute;
  bottom: 0;
  left: -0.5em;
  display: block;
  width: 4.5625em;
  height: 1em;
  background: #FFFFFF;
  box-shadow: 0 0.4375em 0 -0.0625em #607D8B;
}

.c0:nth-child(2) {
  z-index: 0;
  background: #607D8B;
  box-shadow: -2.1875em 0.6875em 0 -0.6875em #607D8B, 2.0625em 0.9375em 0 -0.9375em #607D8B, 0 0 0 0.375em #607D8B, -2.1875em 0.6875em 0 -0.3125em #607D8B, 2.0625em 0.9375em 0 -0.5625em #607D8B;
  opacity: 0.6;
  -webkit-transform: scale(0.5) translate(6em,-3em);
  -ms-transform: scale(0.5) translate(6em,-3em);
  transform: scale(0.5) translate(6em,-3em);
  -webkit-animation: klvFln 4s linear infinite;
  animation: klvFln 4s linear infinite;
}

.c0:nth-child(2):after {
  background: #607D8B;
}

<div
  className="c0"
  data-e2e="cloud"
/>
`;
